src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PartialUpdateSpec.groovy

```diff
@@ -89,7 +233,134 @@ class PartialUpdateSpec extends HealthCheckSpecification {
         flowHelperV2.deleteFlow(flow.flowId)
     }
 
-    def "Partial update with empty body does not actually update flow in any way"() {
+    @Tidy
+    def "Able to update a flow endpoint using partial update"() {
+        given: "Three active switches"
+        def allSwitches = topology.activeSwitches
+        assumeTrue("Unable to find three active switches", allSwitches.size() >= 3)
+        def srcSwitch = allSwitches[0]
+        def dstSwitch = allSwitches[1]
+
+        and: "A vlan flow"
+        def flow = flowHelperV2.randomFlow(srcSwitch, dstSwitch, false)
+        flowHelperV2.addFlow(flow)
+
+        when: "Update the flow: port number and vlan id on the src endpoint"
+        def flowInfoFromDb1 = database.getFlow(flow.flowId)
+        def newPortNumber = topology.getAllowedPortsForSwitch(topology.activeSwitches.find {
+            it.dpId == flow.source.switchId
+        }).last()
+        def newVlanId = flow.destination.vlanId + 1
+        flowHelperV2.partialUpdate(flow.flowId, new FlowPatchV2().tap {
+            source = new FlowPatchEndpoint().tap {
+                portNumber = newPortNumber
+                vlanId = newVlanId
+            }
+        })
+
+        then: "Flow is really updated"
+        with(northboundV2.getFlow(flow.flowId)) {
+            it.source.portNumber == newPortNumber
+            it.source.vlanId == newVlanId
+        }
+
+        and: "Flow rules are recreated"
+        def flowInfoFromDb2 = database.getFlow(flow.flowId)
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            with(northbound.getSwitchRules(srcSwitch.dpId).flowEntries.findAll {
+                !Cookie.isDefaultRule(it.cookie)
+            }) { rules ->
+                rules.findAll {
```

Comment:
To tell the truth I am afraid when the `qinq` feature is merged this verification can be really difficult
How about checking that flow allows traffic instead of checking specific fields in rules?

I am not insist